assets/training/model_management/src/azureml/model/mgmt/processors/pyfunc/llava/llava_mlflow_wrapper.py

```diff
@@ -0,0 +1,174 @@
+        :return: Pandas dataframe with column ["response"] containing the model's response to the dialog so far.
+        """
+        from llava.constants import IMAGE_TOKEN_INDEX
+        from llava.mm_utils import tokenizer_image_token, KeywordsStoppingCriteria
+
+        # Do inference one input at a time.
+        responses = []
+        for image, prompt, direct_question in zip(
+            input_data[MLflowSchemaLiterals.INPUT_COLUMN_IMAGE],
+            input_data[MLflowSchemaLiterals.INPUT_COLUMN_PROMPT],
+            input_data[MLflowSchemaLiterals.INPUT_COLUMN_DIRECT_QUESTION],
+        ):
+            # Decode the image and make a PIL Image object.
+            pil_image = Image.open(io.BytesIO(process_image(image)))
+
+            # If prompt not specified, make prompt from direct question column.
+            if not prompt:
+                prompt_from_direct_question = True
+                if self._model_version == self.LLAVA_MPT:
+                    prompt = (
+                        f"A conversation between a user and an LLM-based AI assistant. The assistant gives helpful "
+                        f"and honest answers.<|im_end|><|im_start|>user\n<im_start><image><im_end>\n"
+                        f"{direct_question}<|im_end|><|im_start|>assistant"
+                    )
+                elif self._model_version == self.LLAVA_7B:
+                    prompt = (
+                        f"[INST] <<SYS>>\nYou are a helpful language and vision assistant. You are able to understand "
+                        f"the visual content that the user provides, and assist the user with a variety of tasks "
+                        f"using natural language.\n<</SYS>>\n\n<im_start><image><im_end>\n{direct_question} [/INST]"
+                    )
+                elif self._model_version == self.LLAVA_13B:
+                    prompt = (
+                        f"input to llava: A chat between a curious human and an artificial intelligence assistant. "
+                        f"The assistant gives helpful, detailed, and polite answers to the human's questions. USER: "
+                        f"<im_start><image><im_end>\n{direct_question} ASSISTANT:"
+                    )
+            else:
+                prompt_from_direct_question = False
+
+            # Make image input.
+            image_tensor = self._image_processor.preprocess(
+                pil_image, return_tensors="pt"
+            )["pixel_values"].half().cuda()
+
+            # Make text input.
+            input_ids = tokenizer_image_token(
+                prompt, self._tokenizer, IMAGE_TOKEN_INDEX, return_tensors="pt"
+            ).unsqueeze(0).cuda()
+            stopping_criteria = KeywordsStoppingCriteria([self._stop_str], self._tokenizer, input_ids)
+
+            # Call model.
+            output_ids = self._model.generate(
+                input_ids,
+                images=image_tensor,
+                do_sample=True,
+                temperature=0.2,
```

Comment:
Should we add the parameter values for temperature and max_new_tokens as constants in config.py ?

Reply:
Same comment as above.